pkg/controller/subjectpermission/subjectpermission_controller.go

```diff
@@ -133,30 +132,27 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 	}
 
 	if len(instance.Spec.ClusterPermissions) != 0 {
-		// build a clusterRoleBindingNameList which consists of clusterRoleName-subjectName
-		crClusterRoleBindingNameList := buildClusterRoleBindingCRList(instance)
-
-		// check ClusterRoleBindingName
-		populateCrClusterRoleBindingNameList := populateClusterRoleBindingNames(crClusterRoleBindingNameList, clusterRoleBindingList)
-
 		var CreatedClusterRoleBindingCount int
-		// loop through crClusterRoleBindingNameList
-		for _, clusterRoleBindingName := range populateCrClusterRoleBindingNameList {
-			// get the clusterRoleName by spliting the clusterRoleBindng name
-			clusterRBName := strings.Split(clusterRoleBindingName, "-")
-			clusterRoleName := clusterRBName[0]
-			subjectName := clusterRBName[1]
+		// build a clusterRoleNameList
+		var crClusterRoleNameList []string
+		for _, clusterRoleName := range instance.Spec.ClusterPermissions {
+			crClusterRoleNameList = append(crClusterRoleNameList, clusterRoleName)
+		}
+
+		for _, a := range crClusterRoleNameList {
+			// add subjectName to clusterRoleName to make clusterRoleBindingName
+			clusterRoleBindingName := a + instance.Spec.SubjectName
 
 			// create a new clusterRoleBinding on cluster
-			newCRB := newClusterRoleBinding(clusterRoleName, subjectName, instance.Spec.SubjectKind)
+			newCRB := newClusterRoleBinding(a, instance.Spec.SubjectName, instance.Spec.SubjectKind)
 			err := r.client.Create(context.TODO(), newCRB)
 			if err != nil {
 				if k8serr.IsAlreadyExists(err) {
 					continue
 				}
 
 				var clusterRoleNames []string
-				clusterRoleNames = append(clusterRoleNames, clusterRoleName)
+				clusterRoleNames = append(clusterRoleNames, clusterRoleBindingName)
```

Comment:
Is this var the right name? You're appending clusterRoleBinding names to a list called ClusterRole names.